### PR TITLE
feat: Add Typescript typings to Breadcrumb, BreadcrumbItems, and BreadcrumbSkeleton

### LINKED ISSUE #13546: Add TypeScript types to Breadcrumb, BreadcrumbItem, BreadcrumbSkeleton

### PR DESCRIPTION
**Pull request by Tresau**
Closes #13546 

Add typings to Breadcrumb, BreadcrumbItems, and BreadcrumbSkeleton.  During work, discovered incorrect PropType in BreadcrumbItem for 'aria-current' and opened #15034 as a result.

#### Changelog

**New**

- Added Typescript types to Breadcrumb, BreadcrumbItems, BreadcrumbSkeleton

**Changed**

- File extensions from .js to .tsx

**Removed**

- n/a

#### Testing / Reviewing

Verify that the storybooks run properly.
Run unit tests

### CODE REVIEW

**Comment by Tresau-IBM** [comment]
I have read the DCO document and I hereby sign the DCO.

File: packages/react/src/components/Breadcrumb/BreadcrumbItem.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        })}
+      </li>
+    );
+  });
+
+BreadcrumbItem.displayName = 'BreadcrumbItem';
+
+BreadcrumbItem.propTypes = {
+  // @ts-expect-error - v12 TODO: BREAKING: This should match AriaAttributes['aria-current']
+  'aria-current': PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
```
Comment (Tresau) [inline packages/react/src/components/Breadcrumb/BreadcrumbItem.tsx:107]: This definition is to broad for 'aria-current'.  Per my discussion [here](https://ibm-cloudplatform.slack.com/archives/C03C8VASVED/p1697813662395289) I have decided to simply suppress the error in order to not introduce breaking changes in the typing additions.  I have opened #15034 where I will address this issue.

Comment (Tresau) [reply packages/react/src/components/Breadcrumb/BreadcrumbItem.tsx:107]: I have created #15052 to address the Incorrect PropTypes to be accepted at a later time as it is a Breaking change.

**Review by alisonjoseph — APPROVED** [review]

**Review by andreancardona — APPROVED** [review]
thank you!! 🎉

**Pull request merged**